resources/src/Data.tsx

```diff
@@ -1,75 +1,80 @@
-import * as React from "react";
+import { ReactElement, ReactNode, createContext, useEffect, useState } from 'react'
 
 // https://reactjs.org/docs/faq-ajax.html
 // https://reactjs.org/docs/hooks-reference.html#usecontext
 
 enum TranslationStatus {
-    Outdated = 1,
-    Translated = 2,
+  Outdated = 1,
+  Translated = 2,
 }
 
-type OperatingSystem = string;
-type Language = string;
-type PageName = string;
+type OperatingSystem = string
+type Language = string
+type PageName = string
 
 interface TranslationData {
-    last_update: string,
-    languages: Language[],
-    entries: Record<OperatingSystem, TranslationOS>,
+  last_update: string
+  languages: Language[]
+  entries: Record<OperatingSystem, TranslationOS>
 }
 
 interface TranslationOS {
-    progress: Record<Language, number>,
-    pages: Record<PageName, TranslationPage>,
+  progress: Record<Language, number>
+  pages: Record<PageName, TranslationPage>
 }
 
 interface TranslationPage {
-    status: Record<Language, TranslationStatus>,
+  status: Record<Language, TranslationStatus>
 }
 
-const DataContext = React.createContext<TranslationData | null>(null);
-const ErrorMessageContext = React.createContext<string | null>(null);
+const DataContext = createContext<{
+  data: TranslationData | null
+  error: string | null
+}>({ data: null, error: null })
 
 // We're using React.ReactElement & React.ReactNode instead of JSX.Element: https://stackoverflow.com/a/47899926/4106848
-type DataFetcherProps = { error: React.ReactElement, loading: React.ReactElement, children: React.ReactNode };
+type DataFetcherProps = {
+  error: ReactElement
+  loading: ReactElement
+  children: ReactNode
+}
 const DataFetcher = (props: DataFetcherProps) => {
-    const [error, setError] = React.useState<string | null>(null);
-    const [isLoaded, setIsLoaded] = React.useState<boolean>(false);
-    const [data, setData] = React.useState<TranslationData | null>(null);
+  const [error, setError] = useState<string | null>(null)
+  const [isLoaded, setIsLoaded] = useState<boolean>(false)
+  const [data, setData] = useState<TranslationData | null>(null)
+
+  // [] = only run on component mount
```

Comment:
This comment looks a little misplaced — should it go down on line 66 where `[]` is written, or am I incorrect?

Reply:
Yes you're right. It's better placed in line 66 👍